drivers/kw2xrf/kw2xrf_spi.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_kw2xrf
+ * @{
+ * @file        kw2xrf_spi.c
+ * @brief       Implementation of SPI-functions for the kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "kw2xrf.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_internal.h"
+
+#include "cpu-conf.h"
+#include "periph/gpio.h"
+#include "irq.h"
+
+/* SPI1 connected to the modem (ieee802.15.4 radio) */
+#define KW2XDRF_SPI                   SPI1
+#define KW2XDRF_SPI_CLKEN()           (SIM->SCGC6 |= (SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_CLKDIS()          (SIM->SCGC6 &= ~(SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_IRQ               SPI1_IRQn
+#define KW2XDRF_SPI_IRQ_HANDLER       isr_spi1
+
+void kw2xrf_spi_init(void)
+{
+    /* enable clocks */
+    KW2XDRF_SPI_CLKEN();
+    KW2XDRF_PORT_CLKEN();
+
+    /* set pins to AF mode */
+    KW2XDRF_PORT->PCR[KW2XDRF_PCS0_PIN] = PORT_PCR_MUX(KW2XDRF_PIN_AF);
+    KW2XDRF_PORT->PCR[KW2XDRF_SCK_PIN] = PORT_PCR_MUX(KW2XDRF_PIN_AF);
+    KW2XDRF_PORT->PCR[KW2XDRF_SOUT_PIN] = PORT_PCR_MUX(KW2XDRF_PIN_AF);
+    KW2XDRF_PORT->PCR[KW2XDRF_SIN_PIN] = PORT_PCR_MUX(KW2XDRF_PIN_AF);
+    /* enable pullup for sin-pin (miso) */
+    KW2XDRF_PORT->PCR[KW2XDRF_SIN_PIN] |= (PORT_PCR_PE_MASK | PORT_PCR_PS_MASK);
+
+    /* max spi clock rate for read is 9MHz, for write 16MHz */
+    /* set speed for 8-bit access */
+    KW2XDRF_SPI->CTAR[0] = SPI_CTAR_FMSZ(7)
+                           | SPI_CTAR_PBR(0)
+                           | SPI_CTAR_BR(1);
+    /* set speed for 16-bit access */
+    KW2XDRF_SPI->CTAR[1] = SPI_CTAR_FMSZ(15)
+                           | SPI_CTAR_PBR(0)
+                           | SPI_CTAR_BR(1);
+
+    /* enable SPI */
+    KW2XDRF_SPI->MCR = SPI_MCR_MSTR_MASK
+                       | SPI_MCR_PCSIS(1)
+                       | SPI_MCR_DOZE_MASK
+                       | SPI_MCR_CLR_TXF_MASK
+                       | SPI_MCR_CLR_RXF_MASK;
+
+    KW2XDRF_SPI->RSER = (uint32_t)0;
+
+    /* set up IRQ */
+    gpio_init_int(GPIO_KW2XDRF, GPIO_NOPULL, GPIO_FALLING, &kw2xrf_rx_irq, NULL);
```

Comment:
looks strange to have this call inside the `spi_init`, I think I would move it to the general driver init() function